File: 🟡 Medium/2491 - Divide Players Into Teams of Equal Skill/_2491DividePlayersIntoTeamsofEqualSkill.js
/**
 * @param {number[]} skill
 * @return {number}
 */
var dividePlayers = function (skill) {
    skill.sort((a, b) => a - b);

    let l = 0, r = skill.length - 1;
    let start = skill[l] + skill[r];
    let res = 0;

    while (l < r) {
        if (skill[l] + skill[r] != start) {
            return -1;
        }

        res += (skill[l] * skill[r]);
        l++;
        r--;
    }

    return res;
};

File: 🟡 Medium/2491 - Divide Players Into Teams of Equal Skill/_2491DividePlayersIntoTeamsofEqualSkill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(), skill.end());

        int l = 0, r = skill.size() - 1;
        int start = skill[l] + skill[r];
        long long res = 0;

        while (l < r) {
            if (skill[l] + skill[r] != start) {
                return -1;
            }

            res += (skill[l] * skill[r]);
            l++;
            r--;
        }

        return res;
    }
};